File: 388. Longest Absolute File Path/Java/Longest Absolute File Path.java
public class Solution {
    public int lengthLongestPath(String input) {
        Stack<Integer> stack = new Stack<>();
        int maxLength = 0;
        String[] items = input.split("\n");
        for (String item : items) {
            int countTabs = item.lastIndexOf('\t') + 1;
            while (stack.size() > countTabs) {
                stack.pop();
            }
            int length = (stack.size() == 0 ? 0 : stack.peek()) + item.length() - countTabs + 1;
            if (item.contains(".")) {
                maxLength = Math.max(maxLength, length - 1);
            }
            stack.push(length);
        }
        return maxLength;
    }
}

File: 388. Longest Absolute File Path/Python/Longest Absolute File Path.py
class Solution(object):
    def lengthLongestPath(self, input):
        """
        :type input: str
        :rtype: int
        """
        stack = []
        max_length = 0
        for item in input.split('\n'):
            count_tabs = item.rfind('\t') + 1
            while len(stack) > count_tabs:
                stack.pop()
            length = (stack[-1] if stack else 0) + len(item) - count_tabs + 1
            if '.' in item:
                max_length = max(max_length, length - 1)
            stack.append(length)
        return max_length
